src/Commands.pm

```diff
@@ -1431,6 +1437,255 @@ sub cmdConf {
 	}
 }
 
+sub cmdxConf {
+	my ($cmd, $args) = @_;
+	my ($file,$file2,$found,$key,$oldval,$shopname,$type,$value, $name, %inf_hash, %ctrl_hash);
+	if ($cmd eq 'sconf') {
+		($key, $value) = $args =~ /(name)(?:\s)(.*)/;
+		$shopname = 1 if $key eq 'name';
+	}
+	($key, $value) = $args =~ /([\s\S]+?)(?:\s)([\-\d\.]+[\s\S]*)/ if !$shopname;
+	$key = $args if !$key;
+	$key =~ s/^\s+|\s+$//g;
+	debug "extracted from args: KEY: $key, VALUE: $value\n";
+	if (!$key) {
+		error "Syntax Error in function '$cmd'. Not found <key>\n".
+				"Usage: $cmd <key> [<value>]\n";
+		return;
+	}
+	if ($cmd eq 'iconf') {
+		%inf_hash  = %items_lut;
+		%ctrl_hash = %items_control;
+		$file = 'items_control.txt';
+		$file2 = 'tables\..\items.txt';
+		$type = 'Item';
+	} elsif ($cmd eq 'mconf') {
+		%inf_hash  = %monsters_lut;
+		%ctrl_hash = %mon_control;
+		$file = 'mon_control.txt';
+		$file2 = 'tables\..\monsters.txt';
+		$type = 'Monster';
+	} elsif ($cmd eq 'pconf') {
+		%inf_hash  = %items_lut;
+		%ctrl_hash = %pickupitems;
+		$file = 'pickupitems.txt';
+		$file2 = 'tables\..\items.txt';
+		$type = 'Item';
+	} elsif ($cmd eq 'sconf') {
+		%inf_hash = %items_lut;
+		%ctrl_hash = %shop;
+		$file = 'shop.txt';
+		$file2 = 'tables\..\items.txt';
+		$type = 'Item';
+	}
+	## Command "sconf" don't have setall & clearall feature
+	if( (($key eq "clearall") || ($key eq "setall")) && ($cmd eq 'sconf')) {
+		error "Syntax Error in function '$cmd'. Keys 'setall' and 'clearall' is not suported.\n";
+		return;
+	} elsif($key eq "clearall")	{ ## If $key is "clear" clear file content and exit
+		fileclear($file);
+		return;
+	} elsif (($key eq "setall")) { ## If $key is "setall" setting all keys in file to $key
+		filesetall($file, $value);
+		return;
+	}
+	## Check $key in tables\monsters.txt & tables\items.txt
+	if ($key ne "all") {
+
+		#key is an ID, have to find the name of the item/monster
+		if ($inf_hash{$key}) {
```

Comment:
Continuing the argument of using references, you would access this as `${$inf_hash}{$key}`